### PR TITLE
Move AccountCheckException to public api

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Missing core features as required by https://github.com/owncloud/password_policy/pull/15

## Related Issue
https://github.com/owncloud/password_policy/pull/15

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -893,7 +894,9 @@ protected function loginUser($user, $password) {
 			}
 
 			return true;
-		}, ['before' => ['uid' => $user, 'password' => $password], 'after' => ['uid' => $user, 'password' => $password]], 'user', 'login');
+		}, ['before' => ['user' => $user, 'uid' => $uid, 'password' => $password],
```
Comment (butonic) [inline lib/private/User/Session.php]: can you fire the events in here (and maybe in other methods in this class) in the flow of the code as in `loginWithPassword`
https://github.com/owncloud/core/blob/0e38d3a88d1c46e54036aaa8fd5e0eaeac895201/lib/private/User/Session.php#L499-L533

That would make the code a lot more readable.

Comment (DeepDiver1975) [reply lib/private/User/Session.php]: EventEmitterTrait::emittingCall was developed on purpose to be used as general design pattern to emit before and after events without interfering the regular code flow.

Comment (butonic) [reply lib/private/User/Session.php]: which is the most stupid idea ever ... it makes the code completely unreadable. Even worse it leads to references like `&$user, &$password` It is too hard to understand. Get rid of it!

Comment (butonic) [reply lib/private/User/Session.php]: in a different PR

**Review by butonic — APPROVED** [review]

**Comment by PVince81** [comment]
@DeepDiver1975 please backport to stable10, will be p2 as required by password policy

**Pull request merged**